### DISCUSSION TITLE
Why are the inelastic models inactivated after cracking in smeared cracking model?

### DISCUSSION THREAD

**Discussion by xuxiaobei1995** [idaholab/moose#22894] [Q&A Modules: Solid mechanics] [answered]
Hi all,

Recently I'm doing some work concerning the cracking. I read the codes in ComputeSmearedCrackingStress and find that the inelastic models are inactivated after the cracking happens. Does any one know the reason of this operation? How can I do to keep the inelastic models activated after cracking?

Best Regards,
Xiaobei

**Comment by GiudGiud** [comment]
@bwspenc I think you wrote this object

**Comment by hugary1995** [comment, marked as answer]
> Recently I'm doing some work concerning the cracking. I read the codes in ComputeSmearedCrackingStress and find that the inelastic models are inactivated after the cracking happens. Does any one know the reason of this operation?

There are many types of inelastic(plastic)-damage models:
E-D
E-D-P
E-D-PD
E-P-D
E-P-PD
where E, P, D, PD divide the state of the material into 4 categories
E: purely elastic phase, little damage growth, no plastic flow
P: purely plastic phase, mostly plastic deformation, small elastic deformation, little damage growth
D: purely softening phase, a lot of damage growth, little plastic flow
PD: a mix of plastic flow and damage softening

We happen to have implemented the E-P-D type, that's it. I believe there were good reasons for that choice.

> How can I do to keep the inelastic models activated after cracking?

Not that I know from an input file, no. But that `ComputeMultipleInelasticStress` or its alike is a complicated beast, there might be some magic parameter that allows you to do that.

If you are willing to modify the source, yes.

**Reply by xuxiaobei1995** [reply]
Hi hugary, In fact I found that the smeared cracking may not be suitable for my work. Anyway, thanks for your explanation!
